fiftyone/utils/complex_yolo/zoo.py

```diff
@@ -44,250 +75,200 @@ class TorchComplexYOLOv3ModelConfig(
     arguments.
 
     Args:
+        config_url: a URL from which the model's config can be downloaded,
+            if necessary
         config_base_filename: the filename in ``fo.config.model_zoo_dir`` in
             which to store the model's config file
-        config_base_url: a URL from which the model's config can be downloaded,
-            if necessary
-        weights_base_filename: the filename in ``fo.config.model_zoo_dir`` in
-            which to store the model's weights file
-        weights_base_gid: a google drive ID from which the model's weights can be
-            downloaded, if necessary
-        classes (None): an optional list of custom classes to use for zero-shot
-            prediction
+        nms_thresh (0.3) a non-maximum suppression threshold to use
     """
 
     def __init__(self, d):
         d = self.init(d)
         super().__init__(d)
 
+        self.config_url = self.parse_string(d, "config_url")
         self.config_base_filename = self.parse_string(
             d, "config_base_filename"
         )
-        self.config_base_url = self.parse_string(d, "config_base_url")
-        self.weights_base_filename = self.parse_string(
-            d, "weights_base_filename"
-        )
-        self.weights_base_gid = self.parse_string(d, "weights_base_gid")
-        self.weights_path = os.path.join(
-            fo.config.model_zoo_dir, self.weights_base_filename
-        )
-        self.config_path = os.path.join(
+        self.nms_thresh = self.parse_number(d, "nms_thresh", default=0.3)
+
+        self._config_path = os.path.join(
             fo.config.model_zoo_dir, self.config_base_filename
         )
-        self.device = device
-        self.Tensor = Tensor
-        self.labels = d["classes"].replace(" ", "").split(",")
+
+    @property
+    def config_path(self):
+        return self._config_path
+
+    def download_config_if_necessary(self):
+        if not os.path.isfile(self._config_path):
+            logger.info("Downloading YOLOv3 config...")
+            etaw.download_file(self.config_url, path=self._config_path)
 
 
 class TorchComplexYOLOv3Model(fout.TorchImageModel):
-    """Torch implementation of Complex-YOLOv3 from https://github.com/ghimiredhikura/Complex-YOLOv3.
+    """Torch implementation of Complex-YOLOv3 from
+    https://github.com/ghimiredhikura/Complex-YOLOv3.
 
     Args:
         config: a :class:`TorchComplexYOLOv3ModelConfig`
     """
 
-    def __init__(self, config, args=dict()):
-        self._device = device
-        self.img_size = args.get("img_size", 608)
-        self.nms_thresh = args.get("nms_thresh", 0.3)
-        self.conf_thresh = args.get("conf_thresh", 0.9)
+    def __init__(self, config):
```

Comment:
`config` should define all parameters that the model can accept. No additional `args` here.